### PR TITLE
Fix official sample

### PR DESCRIPTION
**Pull request by jdeantoni**
second attempt :)

this is an alignment of "official" concurrent example with the last gemoc release

### CODE REVIEW

**Comment by jdeantoni** [comment]
well...
Erwan was right I used a "git config --global user.email="julien.deantoni@univ-cotedazur.fr" unexpectedly...

So I updated it in the gemoc git folder with the correct identity.
I closed previous PR et did a new one... It seems to be the same problem (maybe because one of the commit still have the wrong committer ID).

@dvojtise not sure to know what is this squash about but If you can accept it when you have time that's perfect. No rush...

I'll try to take care about my git identity in the future....

**Comment by dvojtise** [comment]
the squash an commit is one of the possible action when accepting a pull request:
- merge, will take all commits and add all of them them to the history
- squash, rewrite the commits into a single one, doing this implies to rewrite the commit messages to synthesise all the other one. It is them possible to fix the wrong or missing signature in a single commit if I'm sure that you have the ECA :smile: 

We use squash for most of the GEMOC PR as it creates a cleaner history

**Comment by ebousse** [comment]
@dvojtise if I am correct @jdeantoni would simply need to edit the description of the PR and to add the "Signed off" at the end of it? Then when doing the squash, the comment of the PR would become the description of the single resulting commit, which would thus contain the "Signed off" declaration?

**Comment by ebousse** [comment]
@jdeantoni I just loaded the TFSM example in a workspace and I get this compilation error for the 
`org.eclipse.gemoc.example.moccml.tfsm.design` project:
```
Description	Resource	Path	Location	Type
The method getcurrentState(EObject) is undefined for the type OncurrentTFSMRTDAccessor	K3RuntimeDataHelper.java	/org.eclipse.gemoc.example.moccml.tfsm.design/src/org/eclipse/gemoc/example/moccml/tfsm/design/services	line 12	Java Problem
```

**Comment by dvojtise** [comment]
@ebousse I don't think the ECA test is based on PR comment, I think they are based on commit comments, this means that correctly fixing this would require to rewrite the git history and amend the commit comments

Note that when doing the squash: the default commit message proposed by the web UI is a concat of all commit messages (ie. with a lot of signed-of-by  that need to be removed and make sure to keep only the last correct one) (editing the PR description has no impact on this)

**Comment by ebousse** [comment]
@dvojtise Yes I see, I thought that the PR message was used as the default commit message, but I mixed things up. You are right.

**Comment by ebousse** [comment]
@jdeantoni I am struggling to make the TFSM example work. I found three problems so far:
- the name of the plugin `org.eclipse.gemoc.example.moccml..tfsm` has an extra dot (not a big deal)
- the value of`toCCSLQVTOFilePath` is not set in the `plugin.xml` file of `org.eclipse.gemoc.example.moccml..tfsm`, it needs to be `/org.eclipse.gemoc.example.moccml.tfsm.dse/qvto-gen/modeling/TFSM.qvto`
- `TFSM.ecl` seems to have some problems: first the imports are still using `org.gemoc` and not `org.eclipse.gemoc`, second I have a compilatio error on line 28 (`res` is not recognized)

**Comment by ebousse** [comment]
Regarding SigPML, I managed to make a model execution work! (although without animation for some reason, probably unrelated to the DSL definition itself). 

A few comments on things I have seen:
- you can remove guava from the dependencies of `org.eclipse.gemoc.example.moccmlsigpml.model`,
- you can probably avoid versionning `org.eclipse.gemoc.example.moccmlsigpml.model/bin`, and add it in the gitignore,
- in `examples/moccmlSigPML/language_workbench/org.eclipse.gemoc.example.moccmlsigpml.model/.project` you can remove the project natures `fr.obeo.dsl.viewpoint.nature.modelingproject` and `org.eclipse.emf.ecoretools.registration.ui.ecoreAutoRegisterNature` which point to nothing in today's eclipse versions,
- in `examples/moccmlSigPML/modeling_workbench/org.eclipse.gemoc.example.moccmlsigpml.simple_example/.project` you can remove the xtext nature I think, and replace it with `org.eclipse.sirius.nature.modelingproject` since SigPML is graphical.

**Comment by ebousse** [comment]
I made a few commits to your branch to remove all remaining `*.class` files from the repository, as it made the review of the diff of the PR not always easy to read. I think a lot of these were pushed a very long time ago by mistake.

File: examples/moccmlTFSM/org.eclipse.gemoc.example.moccml..tfsm/src/org/eclipse/gemoc/example/moccml/tfsm/Activator.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package org.eclipse.gemoc.example.moccml.tfsm;
+
+public class Activator {
```
Comment (ebousse) [inline examples/moccmlTFSM/org.eclipse.gemoc.example.moccml..tfsm/src/org/eclipse/gemoc/example/moccml/tfsm/Activator.java]: Why an activator, if it remains empty?

File: examples/moccmlTFSM/org.eclipse.gemoc.example.moccml.tfsm.design/pom.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ -->
+
+<!-- Copyright (C) 2013 Obeo Cedric Brun <cedric.brun@obeo.fr> -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.gemoc.sample.robotml</groupId>
```
Comment (ebousse) [inline examples/moccmlTFSM/org.eclipse.gemoc.example.moccml.tfsm.design/pom.xml]: Do we still need RobotML for the TFSM build? (I haven't checked but I wonder)

**Comment by ebousse** [comment]
And I rapidly did the same cleanup for `xtend-gen` files. Sorry for making these changes unrelated to the original PR, but it helped me understand the actual changes better + to eliminate the possible cause of a bug.

As shown above, I tried using the "code review" tool of github, but the diff is very long and it slows down my firefox way too much, so I only went through a couple of files :)

**Comment by dvojtise** [comment]
should be handled by https://github.com/eclipse/gemoc-studio-execution-moccml/pull/19

**Comment by dvojtise** [comment]
(I moved the tfsm sample to a structure similar to other examples ie. language_workbench subfolder (in order to have room for model examples) )


when I tried to open the projects I have an error on K3RuntimeDataHelper which apprently rely on a generated file org.eclipse.gemoc.example.moccml.tfsm/xdsml-java-gen/oncurrenttfsm/xdsml/api/impl/OncurrentTFSMRTDAccessor.java

but with my version of GEMOC it doesn't generate the expected methods (getcurrentState and getnumberOfTicks) 

Did you forget to do a PR that contributes something to the generator ? (most probably in org.eclipse.gemoc.execution.concurrent.ccsljavaxdsml.ui/src/main/java/org/eclipse/gemoc/execution/concurrent/ccsljavaxdsml/ui/builder/GemocLanguageDesignerBuilder.java ? )

**Comment by dvojtise** [comment]
I think I've fixed TFSM which seem to work now

I've also done some minor corrections in SigPML

I've also fixed the example deployer plugin so it is now in the studio with both TFSM and SigPML


some bugs remains, but they aren't directly  due to the examples but to other bugs (bug when doing a clean all, breakpoint management which ideally would require some refactoring, or the fact that the project have 2 natures (concurrent and sequential) cf https://github.com/eclipse/gemoc-studio-execution-moccml/issues/24 and https://github.com/eclipse/gemoc-studio-modeldebugging/issues/131 https://github.com/eclipse/gemoc-studio-modeldebugging/issues/132

**Pull request merged**